### PR TITLE
Setup Elliptic systems directory structure

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This PR adds the `Elliptic/Systems/` directory structure with three systems that we plan to implement. Adding this allows @geoffrey4444 and me to start contributing code.

### Code review checklist

- [x] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For instructions on how to perform the CI checks locally refer to the [Dev guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [x] The code is documented and the documentation renders correctly. Run `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`. Then open `index.html`.
- [x] The code follows the stylistic and code quality guidelines listed in the [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — CHANGES_REQUESTED** [review]
Please make this look very similar to #742 
That is add the CMake files, add tests for the names of the tags, introduce an EllipticSystems doxygen group, document the namespaces

**Review by kidder — COMMENTED** [review]
squash these

**Comment by nilsvu** [comment]
👍

**Review by kidder — APPROVED** [review]
@nilsdeppe please look at

File: tests/Unit/Elliptic/Systems/XCTS/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+set(LIBRARY_SOURCES
+  Test_XCTSTags.cpp
+  )
+
+add_test_library(
+  ${LIBRARY}
+  "Elliptic/Systems/XCTS/"
+  "${LIBRARY_SOURCES}"
+  "XCTS"
```
Comment (kidder) [inline tests/Unit/Elliptic/Systems/XCTS/CMakeLists.txt]: this needs to be the empty string `""`until a library is actually created

File: tests/Unit/Elliptic/Systems/ThermalNoise/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+set(LIBRARY_SOURCES
+  Test_ThermalNoiseTags.cpp
+  )
+
+add_test_library(
+  ${LIBRARY}
+  "Elliptic/Systems/ThermalNoise/"
+  "${LIBRARY_SOURCES}"
+  "ThermalNoise"
```
Comment (kidder) [inline tests/Unit/Elliptic/Systems/ThermalNoise/CMakeLists.txt]: `""` until the library exists

File: tests/Unit/Elliptic/Systems/Poisson/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+set(LIBRARY_SOURCES
+  Test_PoissonTags.cpp
+  )
+
+add_test_library(
+  ${LIBRARY}
+  "Elliptic/Systems/Poisson/"
+  "${LIBRARY_SOURCES}"
+  "Poisson"
```
Comment (kidder) [inline tests/Unit/Elliptic/Systems/Poisson/CMakeLists.txt]: `""` until the library exists

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Elliptic/Systems/XCTS/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBRARY XCTS)
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/XCTS/CMakeLists.txt]: `Xcts` both the lib name and the directory. What you currently have is not CamelCase.

File: src/Elliptic/Systems/XCTS/Tags.hpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+
+/*!
+ * \ingroup EllipticSystemsGroup
+ * \brief Items related to solving the Extended Conformal Thin Sandwich (XCTS)
+ * equations.
+ */
+namespace XCTS {
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/XCTS/Tags.hpp]: `xcts` will be much easier to type, but the other allowed alternative is `Xcts`

File: src/Elliptic/Systems/XCTS/Tags.hpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+/// Defines DataBox tags for the Extended Conformal Thin Sandwich (XCTS) system
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/XCTS/Tags.hpp]: You need to `/// \cond` and `/// \endcond` around the forward decls

File: tests/Unit/Elliptic/Systems/XCTS/Test_XCTSTags.cpp [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// Distributed under the MIT License.
```
Comment (nilsdeppe) [inline tests/Unit/Elliptic/Systems/XCTS/Test_XCTSTags.cpp]: `Test_XctsTags.cpp`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
👍

File: src/Elliptic/Systems/Poisson/Tags.hpp [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+/// Defines DataBox tags for the Poisson system
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/Poisson/Tags.hpp]: cond

File: src/Elliptic/Systems/ThermalNoise/Tags.hpp [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+/// Defines DataBox tags for the ThermalNoise system
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/ThermalNoise/Tags.hpp]: cond

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
fixed

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe please look at

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**